bin/phpstan/filtered-functions.stub

```diff
@@ -29,3 +29,57 @@
  * }|null Attachment metadata. Null on failure.
  */
 function wp_get_attachment_metadata( int $attachment_id = 0, bool $unfiltered = false ): ?array {}
```

Comment:
Can we rename stub files to `.php` for syntax highlighting?

Reply:
Also, I recommend contributing these enhancements to https://github.com/php-stubs/wordpress-stubs

Reply:
PHPStan suggests the `.stub` extension:

> [![image](https://github.com/WordPress/performance/assets/134745/0a4ab482-a5f0-4d61-a987-f4997644e1c9)](https://phpstan.org/user-guide/stub-files#:~:text=The%20stub%20file%20needs%20to%20be%20added%20to%20the%20stubFiles%20key%20in%20the%20configuration%20file%3A)

Also, if we go with `stub` then we'll have to fix the PHPCS issues in the file as well, or else ignore it in PHPCS. Happy to do that if you want.

Reply:
> Also, I recommend contributing these enhancements to https://github.com/php-stubs/wordpress-stubs

Actually, these wouldn't be suitable for php-stubs/wordpress-stubs because this is about adding types for the values injected by filters. For example, `wp_get_attachment_metadata()` and `_wp_image_meta_replace_original()` add `dominant_color`, `has_transparency`, and `sources` to the array keys being returned. 

I seem to have included `wp_check_filetype()` by mistake since PHPStan still is happy without it. So I removed it in 2dfc76f85cc55028b4c414755e0888acbbb6fabe.

Reply:
Hmm I see. In that case I wonder if the `wp_get_attachment_metadata` function could be annotated with [generics](https://phpstan.org/blog/generics-by-examples).

It's something I did for `WP_REST_Request` in the past: https://github.com/php-stubs/wordpress-stubs/blob/6e1ef76dcbdca0cf6e1a6079e9d155d456ccec01/functionMap.php#L62-L66